kernel/sched.c

```diff
@@ -681,6 +685,26 @@ void z_sched_start(struct k_thread *thread)
 	z_reschedule(&sched_spinlock, key);
 }
 
+#ifdef CONFIG_SMP
+/**
+ * @brief Test if the CPU is deadlocked waiting on each other in ISRs
+ *
+ * Test if the CPUs are deadlocked waiting on each other in ISRs.
+ *
+ * @return true if deadlock detected, false if not
+ */
```

Comment:
Style, but big doxygen comments on tiny static utilities hurt IMHO more than they help.  This isn't an API, it's just a utility.  Also see below: it's missing the important bit we'd want to see in a comment, which is an explanation of how it works.

Reply:
Very reasonable. Shall do. :)